sqlserver/assets/configuration/spec.yaml

```diff
@@ -287,6 +287,69 @@ files:
           value:
             type: number
             example: 10
+
+    - name: aws
+      description: |
+        Provide the following if you are instrumenting a managed AWS database. Requires `dbm:true`.
+        
+        Requires setting the field `instance_endpoint`, which should be equal to the Endpoint.Address of the instance
+        the agent is connecting to. Note, if the `host` field is already equal to the instance's Endpoint.Address value, 
+        then setting this configuration is not necessary.
+      options:
+        - name: instance_endpoint
+          description: |
+            Equal to the Endpoint.Address of the instance the agent is connecting to.
+            
+            For more information on instance endpoints, 
+            see the AWS docs https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_Endpoint.html
+          value:
+            type: string
+            example: foo.bar.us-east-1.rds.amazonaws.com
+
+    - name: gcp
+      description: |
+        Provide the following if you are instrumenting a managed GCP database. Requires `dbm:true`.
+      options:
+        - name: project_id
+          description: |
+            Equal to the GCP resource's project id.
+            
+            For more information on project ids,
+            See the GCP docs https://cloud.google.com/resource-manager/docs/creating-managing-projects
+          value:
+            type: string
+            example: foo-project
+        - name: instance_id
+          description: |
+            Equal to the GCP resource's instance id.
+            
+            For more information on instance ids,
+            See the GCP docs https://cloud.google.com/sql/docs/mysql/instance-settings#instance-id-2ndgen
+          value:
+            type: string
+            example: foo-database
+
+    - name: azure
+      description: |
+        Provide the following if you are instrumenting a managed Azure database. Requires `dbm:true`.
+      options:
+        - name: product
+          description: |
+            Equal to the managed Azure DB product. 
+            
+            Acceptable values are:
+              - `flexible_server` 
+              - `single_server`
+          value:
+            type: string
+            example: flexible_server
```

Comment:
Is this relevant for SQL Server? I though single server / flexible server were only mysql / postgres?

Reply:
ah yeah, this was dumb copy pasta from the RFC. We will want to know whether its managed or azure sql db, but we can just name the fields differently like you suggested to get that information

Reply:
I'll update to do this instead 
```
azure:
  # docs...
  managed_instance_name: foo-bar

  # docs...
  sql_database_name: fooBar
  ```

Reply:
actually @justiniso what do you think about instead doing something like this https://github.com/DataDog/integrations-core/pull/11982/commits/f91e899609304639326957c8b5ef4802dcf80090

we can make them set the `deployment_type`, which would keep it consistent with what we require for postgres/mysql e.g.: https://github.com/DataDog/integrations-core/pull/11987/files#diff-49ce2a135c95e30effa64ca894cf499810631cee4b59c6f2a05a82ef42c0be2fR455

possible deployment types we want to cover are bounded and described here https://docs.microsoft.com/en-us/azure/azure-sql/azure-sql-iaas-vs-paas-what-is-overview?view=azuresql